obelix/__init__.py

```diff
@@ -279,24 +298,120 @@ def read_data(self, data_path, no_cifs=False):
 class ShonAndMin(Dataset):
     '''
     Shon and Min dataset class.
-    
+
     Attributes:
-        dataframe (pd.DataFrame): DataFrame containing the dataset.
+        dataframe (pd.DataFrame): Cleaned DataFrame containing the dataset.
     '''
 
-    def __init__(self, data_path="/home/leah/leah---OBELiX/data/Shon_and_Min.xlsx"):
+    def __init__(self, data_path="./SM_rawdata"):
         '''
-        Loads the Shon and Min dataset from an Excel file.
+        Loads and cleans the Shon and Min dataset from Sheet2.
         '''
         self.data_path = Path(data_path)
-        df = self.read_data(self.data_path)
-        super().__init__(df)
-        self.dataframe = df
-    
-    def read_data(self, data_path):
-        '''Reads the Shon and Min dataset.'''
-        data = pd.read_excel(data_path)
+        self.data_path.mkdir(parents=True, exist_ok=True)
+
+        self.source_file = Path("/home/leah/leah---OBELiX/data/ao3c01424_si_001.xlsx")
+
+        raw_df = self.read_data(self.source_file, sheet_name="Sheet2")
+
+        cleaned_df = self._clean_data(raw_df)
+
+        super().__init__(cleaned_df)
+
+    def read_data(self, file_path, sheet_name=None):
+        '''Reads specified sheet from the Excel file.'''
+        data = pd.read_excel(file_path, sheet_name=sheet_name)
         return data
 
+    def _clean_data(self, df):
+        '''Processes and cleans the raw dataframe.'''
+
+        df["Ionic Conductivity"] = df["Ionic Conductivity"].apply(normalize_string)
+
+        df["Ionic Conductivity Numeric"] = df["Ionic Conductivity"].apply(convert_scientific_string)
+
+        df["Ionic Conductivity Numeric (S/cm)"] = df.apply(
+            lambda r: convert_to_S_cm(r.get("Raw_unit", ""), r["Ionic Conductivity Numeric"]),
+            axis=1
+        )
+
+        to_drop = []
+        for idx, row in df.iterrows():
+            cond = row["Ionic Conductivity Numeric (S/cm)"]
+            if pd.isna(cond) or cond <= 0 or not (-18 <= np.log10(cond) <= 0):
+                to_drop.append(idx)
+                continue
+            name = row.get("Name", "")
+            if pd.isna(name) or not is_valid_formula(name):
+                to_drop.append(idx)
+                continue
+
+        df_clean = df.drop(index=to_drop).reset_index(drop=True)
+
+        df_clean["log_target"] = np.log10(df_clean["Ionic Conductivity Numeric (S/cm)"])
+
+        return df_clean
```

Comment:
This is the same as `main` in shon_min.py. You can just rename `main` there and import directly, you don't need to copy it here